pkg/rbac/sharedData.go

```diff
@@ -51,36 +59,112 @@ var managedClusterResourceGvr = schema.GroupVersionResource{
 	Resource: "managedclusters",
 }
 
-func (cache *Cache) PopulateSharedCache(ctx context.Context) error {
+func (shared *SharedData) getPropertyTypes(cache *Cache, ctx context.Context) (map[string]string, error) {
+
+	// original query:
+	// select distinct key, jsonb_typeof(value) as datatype FROM search.resources,jsonb_each(data);
+	var selectDs *goqu.SelectDataset
+	//define schema:
+	schemaTable := goqu.S("search").Table("resources")
+	//data expression to get value and key
+	jsb := goqu.L("jsonb_each(?)", goqu.C("data"))
+	//select from these datasets
+	ds := goqu.From(schemaTable, jsb)
+	//select statement with orderby and distinct clause
+	selectDs = ds.Select(goqu.L("key"), goqu.L("jsonb_typeof(?)",
+		goqu.C("value")).As("datatype")).Distinct()
+
+	query, params, sqlerr := selectDs.ToSQL()
+	if sqlerr != nil {
+		klog.Errorf("Error building Search query: %s", sqlerr.Error())
+	}
+
+	klog.V(5).Infof("Query for property datatypes: [%s] ", query)
+	rows, err := cache.pool.Query(ctx, query, params...)
+	if err != nil {
+		klog.Errorf("Error resolving query [%s] with args [%+v]. Error: [%+v]", query, err)
+		return nil, err
+	}
+	defer rows.Close()
+
+	var key string
+	var value string
+	resourceTypeMap := make(map[string]string)
+	for rows.Next() {
+		err = rows.Scan(&key, &value)
```

Comment:
Declare these inside the loop.
FYI, this is a recent bug fix for a similar problem. https://github.com/stolostron/search-v2-api/pull/98
```suggestion
	resourceTypeMap := make(map[string]string)
	for rows.Next() {
		var key, value string
		err = rows.Scan(&key, &value)
```